[ES|QL] TS Command: mv_expand followed by stats when getting from multiple indices causes unexpected partial results

Example query:
```
ts k8s-downsa*,k8s
| mv_expand `client.ip` 
| stats  GIndJeyVxxBh = count(rate(network.total_bytes_in)) by event_log, UXhUADzsJqyi = bucket(@timestamp,1hour)
;
```
Causes
```
    java.lang.AssertionError: unexpected partial results: _clusters={details={(local)={_shards={total=2, failed=2, successful=0, skipped=0}, took=249, indices=k8s-downsa*,k8s, failures=[{node=null, reason={reason=Index 2 out of bounds for length 2, type=array_index_out_of_bounds_exception}, index=k8s, shard=0}], status=partial}}}
    Expected: (null or is <false>)
         but: was <true>
        at __randomizedtesting.SeedInfo.seed([A00043C5D036A18C:28547C1F7ECACC74]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2797)
        at org.elasticsearch.xpack.esql.qa.rest.RestEsqlTestCase.assertNotPartial(RestEsqlTestCase.java:1911)
        at org.elasticsearch.xpack.esql.qa.rest.RestEsqlTestCase.runEsql(RestEsqlTestCase.java:1401)
        at org.elasticsearch.xpack.esql.qa.rest.RestEsqlTestCase.runEsql(RestEsqlTestCase.java:1412)
```

Another failing query
```
ts k8s-downsampled,k8s*,k8s-d* 
| mv_expand `client.ip` 
| mv_expand event_log 
| eval  PmRBmkGLRo = events_received, `network.eth0.tx` = "ujEiVknoQAHOIMdAbD", ZFFkDcBnCVSw = substring(region, 1, 3) 
| stats  xwQJzaFP = max(rate(network.total_bytes_in)), `events_received` = count(last_over_time(network.total_bytes_out)) by dJHEEAGGTSw = bucket(@timestamp,1hour)
```

--- a/docs/changelog/136931.yaml
+++ b/docs/changelog/136931.yaml
@@ -1,0 +1,6 @@
+pr: 136931
+summary: Don't allow MV_EXPAND prior to STATS with TS
+area: ES|QL
+type: bug
+issues:
+ - 136928

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/TimeSeriesAggregate.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/TimeSeriesAggregate.java
@@ -193,6 +193,17 @@ public class TimeSeriesAggregate extends Aggregate {
                     )
                 );
             }
+            // reject `TS metrics | MV_EXPAND ... | STATS ...`
+            if (p instanceof MvExpand mvExpand) {
+                failures.add(
+                    fail(
+                        mvExpand,
+                        "mv_expand [{}] in the time-series before the first aggregation [{}] is not allowed",
+                        mvExpand.sourceText(),
+                        this.sourceText()
+                    )
+                );
+            }
         });
     }
 

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java
@@ -2813,6 +2813,18 @@ public class VerifierTests extends ESTestCase {
             can only be used after STATS when used with TS command"""));
     }
 
+    public void testMvExpandBeforeTSStatsNotAllowed() {
+        assertThat(error("TS test | MV_EXPAND name | STATS max(network.connections)", tsdb), equalTo("""
+            1:11: mv_expand [MV_EXPAND name] in the time-series before the first aggregation \
+            [STATS max(network.connections)] is not allowed"""));
+
+        assertThat(error("TS test | MV_EXPAND name | MV_EXPAND network.connections | STATS max(network.connections)", tsdb), equalTo("""
+            1:28: mv_expand [MV_EXPAND network.connections] in the time-series before the first aggregation \
+            [STATS max(network.connections)] is not allowed
+            line 1:11: mv_expand [MV_EXPAND name] in the time-series before the first aggregation \
+            [STATS max(network.connections)] is not allowed"""));
+    }
+
     private void checkVectorFunctionsNullArgs(String functionInvocation) throws Exception {
         query("from test | eval similarity = " + functionInvocation, fullTextAnalyzer);
     }

